data_structures/mesh.py

```diff
@@ -80,6 +87,16 @@ class Polygon:
     __slots__ = ("vertexLocations", "normal", "center", "area", "materialIndex")
 
     @staticmethod
+    def fromBMeshFace(bmeshFace, centerWeighted = False):
```

Comment:
ah, I had another variant cause you can calc center as median or median weighted with an advanced option

I'll test that again to make sure we need it